cupy_setup_build.py

```diff
@@ -198,6 +184,24 @@
 ]
 
 
+if 1 == int(os.environ.get('CUPY_SETUP_ENABLE_THRUST', 1)):
+    MODULES.append({
+        'name': 'thrust',
+        'file': [
+            ('cupy.cuda.thrust', ['cupy/cuda/cupy_thrust.cu']),
+        ],
+        'include': [
+            'thrust/device_ptr.h',
+            'thrust/sequence.h',
+            'thrust/sort.h',
+        ],
+        'libraries': [
+            'cudart',
+        ],
+        'check_method': build.check_cuda_version,
+    })
+
+
 def convert_modules_for_hip():
```

Comment:
I guess you need to handle this function `convert_modules_for_hip()` when `CUPY_SETUP_ENABLE_THRUST=0`?

Reply:
Thank you for pointing out!

@okuta 
What are these lines?

https://github.com/cupy/cupy/blob/bcf2ec3f09c31042c6e413a7a160fd92f7a3f215/cupy_setup_build.py#L203-L204

Reply:
`convert_modules_for_hip` is called some times.

Reply:
With different number of elements each time?
What's does the magic number 3 mean?

Reply:
OK, it looks 3 is the number of modules `MODULE` will have after this function is called for the first time. (`cuda`, `cusolver` and `thrust`, I guess?)

I think I need to rewrite this function.

Reply:
0836f9cfe25061ad2108a456b01916b56eedb04b